regression/Makefile

```diff
@@ -8,6 +8,7 @@ DIRS = ansi-c \
        goto-instrument-typedef \
        strings-smoke-tests \
        test-script \
+       cbmc-java-inheritance \
```

Comment:
@peterschrammel  I am okay with both, by motivation for a separate directory was that cbmc-java is getting a bit 'crowded', what do you prefer?